### PR TITLE
fix aws completer path on nixos

### PR DESCRIPTION
**Pull request by contrun**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Add a new aws completer path which works on nixos

## Other comments:

...

### CODE REVIEW

**Comment by larson-carter** [comment]
@contrun I believe that this PR works. Just by simply looking at the code. However I am going to need at least 1 more tester. 

Thank you.

**Comment by contrun** [comment]
Can you please give a link on what kind of test I need to add?

**Comment by larson-carter** [comment]
@contrun 

I'm just needing another user to locally clone your changes and verify that they work.

File: plugins/aws/aws.plugin.zsh [hunk 1]
```diff
@@ -92,6 +92,9 @@ else
     # Ubuntu
     elif [[ -e /usr/share/zsh/vendor-completions/_awscli ]]; then
       _aws_zsh_completer_path=/usr/share/zsh/vendor-completions/_awscli
+    # NixOS
+    elif [[ -e "$(dirname "$(dirname $(realpath "$(command -v aws)"))")/share/zsh/site-functions/aws_zsh_completer.sh" ]]; then
+      _aws_zsh_completer_path="$(dirname "$(dirname $(realpath "$(command -v aws)"))")/share/zsh/site-functions/aws_zsh_completer.sh"
```
Comment (mcornella) [inline plugins/aws/aws.plugin.zsh]: This code runs a fork 3 times at the very least, 6 at the worst. Better use [parameter expansion](http://zsh.sourceforge.net/Doc/Release/Expansion.html#Parameter-Expansion): `${pathvar:h}` for dirname, `${pathvar:P}` for realpath. We can also substitute `command -v aws` for the variable `${commands[aws]}` where `$commands` is a variable holding all the executables in `$PATH`.

Putting it all together:
```suggestion
    elif [[ -e "${commands[aws]:P:h:h}/share/zsh/site-functions/aws_zsh_completer.sh" ]]; then
      _aws_zsh_completer_path="${commands[aws]:P:h:h}/share/zsh/site-functions/aws_zsh_completer.sh"
```

In theory this is the same, test that it works though. BTW, is `realpath` necessary?

**Comment by mcornella** [comment]
Thanks @contrun!

File: plugins/aws/aws.plugin.zsh (hunk 1 above)
Comment (contrun) [reply plugins/aws/aws.plugin.zsh]: No, `realpath` seems to be unnecessary. I didn't check that carefully. Thanks for your suggestion. I surely learned some new tricks. I was just trying to edit it on my editor to delete the `:P` part, this PR is been merged too fast. I guess there is no harm in leaving `:P`. Thank you.

Comment (mcornella) [reply plugins/aws/aws.plugin.zsh]: Oh sorry. Well it doesn't break anything, I'm sure. Might remove it on a later change. Thanks!

Comment (contrun) [reply plugins/aws/aws.plugin.zsh]: No problem. Thanks.

**Pull request merged**